BaragonCore/src/main/java/com/hubspot/baragon/models/UpstreamInfo.java

```diff
@@ -110,16 +127,24 @@ public boolean equals(Object o) {
     if (!originalPath.equals(that.originalPath)) {
       return false;
     }
+    if (!group.equals(that.group)) {
+      return false;
+    }
 
     return true;
   }
 
+  public boolean sameAs(UpstreamInfo that) {
+    return upstream.equals(that.upstream) && group.equals(that.group);
+  }
+
```

Comment:
this is mostly just stylistic, but I'd personally make this a static method named something like `UpstreamInfo.upstreamsAndGroupMatch(UpstreamInfo a, UpstreamInfo b)`